sdk/helper/certutil/helpers.go

```diff
@@ -446,18 +446,30 @@ func ParsePublicKeyPEM(data []byte) (interface{}, error) {
 	return nil, errors.New("data does not contain any valid public keys")
 }
 
-// addPolicyIdentifiers adds certificate policies extension
-//
+// AddPolicyIdentifiers adds certificate policies extension, based on CreationBundle
 func AddPolicyIdentifiers(data *CreationBundle, certTemplate *x509.Certificate) {
-	for _, oidstr := range data.Params.PolicyIdentifiers {
-		oid, err := StringToOid(oidstr)
+	oidOnly := true
+	for _, oidStr := range data.Params.PolicyIdentifiers {
+		oid, err := StringToOid(oidStr)
 		if err == nil {
 			certTemplate.PolicyIdentifiers = append(certTemplate.PolicyIdentifiers, oid)
 		}
+		if err != nil {
+			oidOnly = false
```

Comment:
I may be way off the mark here but I would have expected us to store non-oid parseable values into a separate list that would then be fed into the `CreatePolicyInformationExtensionFromStorageStrings` call below?

Is there not a chance that we could have some of the same elements now within certTemplate.PolicyIdentifiers and within certTemplate.ExtraExtensions?

Reply:
Theoretically (though not practically) possible.  If someone had in their policy identifier extensions, before validation was added, the fields {"oid":"2.5.1.4.8"}, and managed to build this with an old version of go, that might be trimmed to numbers, and possibly successfully parsed as a oid, and could be successfully parsed as a policy json object now.

It's not a problem if some of the same elements are in the `certTemplate.PolicyIdentifiers` and `certTemplate.ExtraExtensions[policyInformationExtension]`.  Everything in `certTemplate.PolicyIdentifiers` is overwritten when there is a policy extension in extra extensions.  Since the extra extension is only created if it can parse _all_ the policy identifiers, it will include everything in `certTemplate.PolicyIdentifiers` (no one will look there and be misled).